Task2/controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.error.ErrorCode;
+import com.epam.esm.error.ErrorHandler;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.dto.CertificateFilterDto;
+import com.epam.esm.entity.dto.CreateGiftCertificateDto;
+import com.epam.esm.entity.dto.IdDto;
+import com.epam.esm.entity.dto.UpdateGiftCertificateDto;
+import com.epam.esm.service.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/certificates")
+public class GiftCertificateController {
+    private GiftCertificateService giftCertificateService;
+
+    @Autowired
+    public void setGiftCertificateService(GiftCertificateService giftCertificateService) {
+        this.giftCertificateService = giftCertificateService;
+    }
+
+    @GetMapping
+    public Set<GiftCertificate> showCertificates() {
+        return giftCertificateService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public GiftCertificate showCertificates(@PathVariable String id) {
+        return giftCertificateService.findById(id).get();
+    }
+
+    @DeleteMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Set<GiftCertificate> deleteCertificate(@RequestBody IdDto idDto) {
+        return giftCertificateService.delete(idDto.getId());
+    }
+
+    @PutMapping
+    public GiftCertificate updateCertificate(@RequestBody UpdateGiftCertificateDto updatedMessage) {
+        return giftCertificateService.update(updatedMessage).get();
+    }
+
+    @PostMapping
+    public Optional<GiftCertificate> addCertificate(@RequestBody CreateGiftCertificateDto newCertificate) {
+        return giftCertificateService.create(newCertificate);
+    }
+
+    @PostMapping("/filter")
```

Comment:
I was thinking that because this operation requires several parameters among which I have 2 arrays for search and sort algorithms it's better to use it as post operation. Especially because potentially this method will have a lot more parameters in it(search by several tags, search by date and other variations)

Reply:
+ Close